src/SAML2/XML/AbstractXMLElement.php

```diff
@@ -43,7 +44,7 @@ public function __toString(): string
      */
     public function serialize(): string
     {
-        return $this->toXML()->ownerDocument->saveXML();
+        return $this->__toString();
```

Comment:
I have reverted this line, because it was breaking tests:
https://travis-ci.org/simplesamlphp/saml2/jobs/643135861#L329

Reply:
Uhm, it's true that the string representation is formatted, while the serialized one should not. Still, in this case I think we should avoid the document type declaration at the top, and this way we won't, since we are not specifying the root element to print.

Reply:
OK so I've basically made a copy of toString, but without the formatting line... Is that what you were saying?